docs/swagger/spec.yml

```diff
@@ -314,6 +314,121 @@ paths:
       responses:
         200:
           description: SUCCESS
+  /uploads/:
+    get:
+      summary: Get a list of uploads
+      description: |
+        The uploads API endpoint enables searching, listing, and creating new uploads.
+      tags:
+        - Imagery
+      parameters:
+        - $ref: '#/parameters/orderingBase'
+        - $ref: '#/parameters/pageSize'
+        - $ref: '#/parameters/page'
+        - $ref: '#/parameters/organization'
+        - $ref: '#/parameters/datasource'
+      responses:
+        200:
+          description: Paginated list of uploads
+          schema:
+            $ref: '#/definitions/UploadPaginated'
+        default:
+          description: Unexpected error
+          schema:
+            $ref: '#/definitions/Error'
+    post:
+      summary: Create an upload
+      description: |
+        Create a new upload.
+      tags:
+        - Imagery
+      parameters:
+        - name: Upload
+          in: body
+          schema:
+            $ref: '#/definitions/UploadCreate'
+      responses:
+        201:
+          description: Upload created
+          schema:
+            $ref: '#/definitions/Upload'
+        default:
+          description: Unexpected error
+          schema:
+            $ref: '#/definitions/Error'
+  /uploads/{uuid}/:
+    get:
+      summary: Retrieve upload details
+      tags:
+        - Imagery
+      parameters:
+        - $ref: '#/parameters/uuid'
+      responses:
+        200:
+          description: Info about upload
+          schema:
+            $ref: '#/definitions/Upload'
+        404:
+          description: |
+            UUID parameter does not refer to an upload or the user is not able to view the upload it refers to
+          schema:
+            $ref: '#/definitions/Error'
+    put:
+      summary: Update an upload
```

Comment:
After it has completed - probably not.

Ah I think I probably need to adjust the `ENUM` so this makes more sense. The flow I imagine is something along the lines of:
 1. User creates upload object (`CREATED`)
 2. User gets credentials, uploads to S3 (`UPLOADING`)
 3. User updates status to indicate no new files are being uploaded and the files to import (e.g where they ended up in s3) (some status to indicated done uploading `UPLOADED`?)
 
When the status is changed to `UPLOADED` a process can be kicked off to process the data then.